src/js/actions/PlayerActions.js

```diff
@@ -55,16 +55,33 @@ const previous = () => {
     });
 };
 
-const shuffle = () => {
+const shuffle = (shuffle) => {
+
+    app.config.set('audioShuffle', shuffle);
+    app.config.saveSync();
+
     store.dispatch({
-        type : AppConstants.APP_PLAYER_SHUFFLE
+        type : AppConstants.APP_PLAYER_SHUFFLE,
+        shuffle
+    });
+};
+
+const repeat = (repeat) => {
+
+    app.config.set('audioRepeat', repeat);
+    app.config.saveSync();
+
+    store.dispatch({
+        type : AppConstants.APP_PLAYER_REPEAT,
+        repeat
     });
 };
 
 const setVolume = (volume) => {
+
     if(!isNaN(parseFloat(volume)) && isFinite(volume)) {
 
-        Player.setAudioVolume(volume);
+        Player.setAudioVolume(Math.pow(volume, 4));
 
         app.config.set('audioVolume', Math.pow(volume, 4));
```

Comment:
Also, what do you think of refactoring this method to dispatch a `APP_PLAYER_VOLUME_CHANGE`, and put this logic into a reducer?